Client/Frontend/Browser/Tab Tray/Views/TabSyncHeaderView.swift

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import BraveUI
+
+protocol TabSyncHeaderViewDelegate {
+    func toggleSection(_ header: TabSyncHeaderView, section: Int)
+}
+
+class TabSyncHeaderView: UITableViewHeaderFooterView, TableViewReusable {
+    
+  var delegate: TabSyncHeaderViewDelegate?
+  var section = 0
+  var collapsed = false
```

Comment:
nit: `collapsed` -> `isCollapsed`, can be marked `private`

Reply:
Fixed at https://github.com/brave/brave-ios/pull/5809/commits/8f2607664aadd1db8ecadc85dd584cb748433a79